core/services/src/main/java/org/exoplatform/services/attachments/storage/AttachmentStorageImpl.java

```diff
@@ -16,42 +16,112 @@
+                                                                                                        entityType.toUpperCase());
+    Utils.sortAttachmentsByDate(attachmentsContextEntity);
+    List<Attachment> attachments = new ArrayList<>();
+    if (!attachmentsContextEntity.isEmpty()) {
+      Session session = null;
+      try {
+        session = Utils.getSession(sessionProviderService, repositoryService);
+        for (AttachmentContextEntity attachmentContextEntity : attachmentsContextEntity) {
+          Attachment attachment = EntityBuilder.fromAttachmentNode(repositoryService,
+                                                                   documentService,
+                                                                   linkManager,
+                                                                   Utils.getCurrentWorkspace(repositoryService),
+                                                                   session,
+                                                                   attachmentContextEntity.getAttachmentId());
+          attachments.add(attachment);
+        }
+      } catch (Exception e) {
+        throw new ObjectNotFoundException("Can't convert attachment JCR node to Attachment DTO");
+      } finally {
+        if (session != null) {
+          session.logout();
+        }
+      }
+    }
+    return attachments;
   }
 
   @Override
-  public AttachmentContextEntity getAttachmentItemByEntity(long entityId, String entityType, String attachmentId) {
-    return attachmentDAO.getAttachmentItemByEntity(entityId, entityType.toUpperCase(), attachmentId);
+  public Attachment getAttachmentItemByEntity(long entityId,
+                                              String entityType,
+                                              String attachmentId) throws ObjectNotFoundException {
+    AttachmentContextEntity attachmentEntity = attachmentDAO.getAttachmentItemByEntity(entityId,
+                                                                                       entityType.toUpperCase(),
+                                                                                       attachmentId);
+    Attachment attachment = null;
+    Session session = null;
+    try {
+      session = Utils.getSession(sessionProviderService, repositoryService);
+      attachment = EntityBuilder.fromAttachmentNode(repositoryService,
+                                                    documentService,
+                                                    linkManager,
+                                                    Utils.getCurrentWorkspace(repositoryService),
+                                                    session,
+                                                    attachmentEntity.getAttachmentId());
+    } catch (Exception e) {
+      throw new ObjectNotFoundException("Can't convert attachment JCR node to Attachment DTO");
```

Comment:
not compatible exceptions, can't transform all `Exception` => `ObjectNotFoundException`